src/rest/CommunityEndpoints.js

```diff
@@ -1,24 +1,200 @@
+    //     blockFreezePeriodSeconds: 0,
+    //     adminFee: 0,
+    //     ...this.options,
+    //     ...options,
+    // }
+
+    await throwIfNotContract(wallet.provider, tokenAddress, 'options.tokenAddress')
+    await throwIfBadAddress(streamrNodeAddress, 'options.streamrNodeAddress')
+    await throwIfBadAddress(streamrOperatorAddress, 'options.streamrOperatorAddress')
+
+    if (adminFee < 0 || adminFee > 1) { throw new Error('options.adminFeeFraction must be a number between 0...1, got: ' + adminFee) }
+    const adminFeeBN = new BigNumber((adminFee * 1e18).toFixed()) // last 2...3 decimals are going to be gibberish
+
+    const stream = await this.createStream({
+        name: `Join-Part-${wallet.address.slice(0, 10)}-${Date.now()}`
+    })
+    debug(`Stream created: ${JSON.stringify(stream.toObject())}`)
+    const res1 = await stream.grantPermission('read', null)
+    debug(`Grant read permission response from server: ${JSON.stringify(res1)}`)
+    const res2 = await stream.grantPermission('write', streamrNodeAddress)
+    debug(`Grant write permission response to ${streamrNodeAddress} from server: ${JSON.stringify(res2)}`)
+
+    const deployer = new ContractFactory(CommunityProduct.abi, CommunityProduct.bytecode, wallet)
+    const result = await deployer.deploy(streamrOperatorAddress, stream.id, tokenAddress, blockFreezePeriodSeconds, adminFeeBN)
+    const { address } = result // this can be known in advance
+    debug(`Community contract @ ${address} deployment started`)
+
+    // add the waiting method so that caller can await community being operated by server (so that EE calls work)
+    const client = this
+    result.isReady = async (pollingIntervalMs, timeoutMs) => client.communityIsReady(address, pollingIntervalMs, timeoutMs)
+    return result
+}
+
+/**
+ * Await this function when you want to make sure a community is deployed and ready to use
+ * @param {EthereumAddress} address of the community
+ * @param {Number} pollingIntervalMs (optional, default: 1000) ask server if community is ready
+ * @param {Number} timeoutMs (optional, default: 60000) give up
+ * @return {Promise} resolves when community server is ready to operate the community (or fails with HTTP error)
+ */
+export async function communityIsReady(address, pollingIntervalMs, timeoutMs) {
+    let stats = await get(this, address, '/stats')
+    const timeout = timeoutMs || 60000
+    const startTime = Date.now()
+    while (stats.error && Date.now() < startTime + timeout) {
+        debug(`Waiting for community ${address} to start. Status: ${JSON.stringify(stats)}`)
+        await sleep(pollingIntervalMs || 1000) // eslint-disable-line no-await-in-loop
+        stats = await get(this, address, '/stats') // eslint-disable-line no-await-in-loop
+    }
+    if (stats.error) {
+        throw new Error(`Community failed to start within ${timeout} ms. Status: ${JSON.stringify(stats)}`)
+    }
```

Comment:
hm, so if the timeout is set to 1000ms but the `get` call takes 10000ms+, it won't cause the function to fail or short circuit. Maybe the parameter should be called `retryTimeoutMs` so it can't be interpreted as an actual max timeout, or perhaps the function should use something like `Promise.race` to implement a hard timeout limit:

```JS
await Promise.race([
  sleep(timeoutMs),
  retry(),
])
```